docs/dapps/sdk/cli.md

```diff
@@ -0,0 +1,99 @@
+      // is `<project_root>/artifacts/.deployment.<network_id>.json`.
+      deploymentFile: './deployments/.deployment.devnet.json',
+      // The `confirmations` field is used to specify the number of block confirmations to wait for
+      // after contract deployment. This is an optional config. If it is not specified, it defaults
+      // to 1 for devnet and 2 for testnet and mainnet.
+      confirmations: 1
+    }
+  }
+}
+
+// You must export the `configuration` from the config file
+export default configuration
+```
+
+</p></details>
+
+In most cases, you only need to specify the `networks` config, and the other optional configs can work well by default. But you can also configure these optional configs if needed:
+
+* `sourceDir`: This config is used to specify the directory path for contract code, it is `<project_root>/contracts` by default
+* `artifactDir`: This config is used to specify the directory path for compiled contract artifacts, it is `<project_root>/artifacts` by default
+* `deployToMultipleGroupsInParallel`: If the contract needs to be deployed to multiple groups, this config specifies whether to deploy contracts on each group in parallel, it is `true` by default
+* `deploymentScriptDir`: This config is used to specify the directory path for deployment scripts, it is `<project_root>/scripts` by default
+* `compilerOptions`: You can refer to the code [here](https://github.com/alephium/alephium-web3/blob/3640e46892c7d2f52942447f300d4b21c7166a0c/packages/web3/src/contract/contract.ts#L104) for default compiler options
+  * `ignoreUnusedConstantsWarnings`: Ignore unused constants warnings if this config is enabled
+  * `ignoreUnusedVariablesWarnings`: Ignore unused variables warnings if this config is enabled
+  * `ignoreUnusedFieldsWarnings`: Ignore unused contract fields warnings if this config is enabled
+  * `ignoreUnusedPrivateFunctionsWarnings`: Ignore unused private functions warnings if this config is enabled
+  * `ignoreUpdateFieldsCheckWarnings`: Ignore compiler warnings if contract functions update contract fields but does not have the `updateField` annotation
+  * `ignoreCheckExternalCallerWarnings`: Ignore compiler warnings if public contract functions does not have the `checkExternalCaller` annotation
+  * `errorOnWarnings`: Compiler warnings will be treated as errors if this config is enabled
+* `skipRecompile`: This config is used to specify whether contract code needs to be recompiled when deploying contracts, it is `false` by default
+* `enableDebugMode`: If this config is enabled, it will print out network requests and error stack traces
+* `forceRecompile`: The purpose of this config is to maintain backward compatibility. If this config is disabled and the contract code has already been deployed to the testnet/mainnet without any updates, then no new bytecode will be generated for the contract. This config is disabled by default.
```

Comment:
There are some rules even when this config is enabled, right? E.g. deployed mainnet contracts will update its bytecode